sdk/search/search-documents/package.json

```diff
@@ -119,7 +119,7 @@
     "rimraf": "^3.0.0",
     "sinon": "^9.0.2",
     "ts-node": "^10.0.0",
-    "typescript": "~4.6.0",
+    "typescript": "~4.8.4",
```

Comment:
Which features are we using that rely on this version of TS? Is this the oldest version that supports what we want to do here?

Reply:
The 'infer T extends U' syntax was introduced in 4.7. The upgrade to 4.8 is to maintain parity with the few packages in the monorepo that have upgraded from 4.6, as they've skipped 4.7. 

It should be possible to work around this with 4.6, but I think there's been discussion about keeping TS current across packages.

Reply:
I'd like to get everyone upgraded to latest over the holidays if we can, since 4.9 is going to be out soon

Reply:
I didn't realize the `infer T extends U` syntax is gated that recently. @xirzec the package will completely break for anyone on an older version of TS if we use that feature. That's still super new and if I'm reading it correctly TS 4.6 won't fall off of the DefinitelyTyped support window until early 2024. @dgetu we will probably need to investigate a couple of alternatives here:

1. We could rework the types to avoid infer-extends. I can take a crack at this, and I think it may be possible by just moving `extends` conditions outside of the `infer` clause.
2. We can use `typesVersions` if downlevel-dts supports this scenario, but customers using TypeScript older than 4.7 (>=4.2), which is the oldest version supported by DefinitelyTyped) will get a sub-par experience that we'd want to make sure is still acceptable.

Reply:
If it becomes too messy to avoid, I'm ok putting a note in the Search README that the best experience requires TS 4.7 in the "querying with TypeScript" section: https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/search/search-documents#querying-with-typescript

I think it's reasonable that we do the best we can with the language version you have, but the best support will always require the newest TS.

Reply:
Moving the `extends` conditions out of the `infer` clauses fixes compatibility for versions >= 4.1.